homeassistant/components/tuya/climate.py

```diff
@@ -141,7 +141,7 @@ def __init__(
 
         # If both temperature values for celsius and fahrenheit are present,
         # use whatever the device is set to, with a fallback to celsius.
-        prefered_temperature_unit = None
+        prefered_temperature_unit = TEMP_CELSIUS
```

Comment:
This change makes no sense IMHO. It will already fall back to whatever is available if `prefered_temperature_unit` is not set. See lines 175 & 191.